gnd/src/main/java/com/google/android/gnd/repository/local/LocalDataStore.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.repository.local;
+
+import com.google.android.gnd.vo.Feature;
+import io.reactivex.Single;
+
+/**
+ * Provides access to local persistent data store. This is the app's canonical store for latest
+ * state; local changes are saved here before being sync'ed with remote stores, and remote updates
+ * are persistent here before being made visible to the user.
+ */
+public interface LocalDataStore {
+
+  /**
+   * Adds a new Feature to the local db and queues the Feature to be added to the remote datastore.
+   *
+   * @param feature the new feature to be added. If set, the feature's ids will be ignore.
```

Comment:
Nit: "The Feature's IDs, if set, will be ignored." So it's clear what "if set" refers to.

Reply:
Done.